ballerina/modules/differ/differ.bal

```diff
@@ -0,0 +1,428 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql_schema_registry.parser;
+
+public isolated function diff(parser:__Schema newSchema, parser:__Schema? oldSchema) returns SchemaDiff[]|Error {
+    return check getDiff(newSchema, oldSchema ?: parser:createSchema());
+}
+
+public isolated function getMajorSeverity(DiffSeverity[] diffs) returns DiffSeverity? {
+    if diffs.length() == 0 {
+        return ();
+    }
+    DiffSeverity majorSeverity = SAFE;
+    foreach DiffSeverity diff in diffs {
+        if diff is BREAKING {
+            majorSeverity = BREAKING;
+            break;
+        }
+        if diff is DANGEROUS {
+            majorSeverity = DANGEROUS;
+        }
+    }
+    return majorSeverity;
```

Comment:
It can be replaced with the following,
```ballerina
from var severity in diffs
order by severity ascending
limit 1
select severity;
```
But the problem is the exact ordering of the enum `DiffSeverity`.
```ballerina
enum DiffSeverity {
    BREAKING,
    DANGEROUS,
    SAFE
}
```
I think the reason that the above query works is because that `BREAKING`, `DANGEROUS` and `SAFE` words are in order alphabetically. I feel like relying on that only is not good. But, I might be wrong. Is there any way in Ballerina to ensure the ordering of the `enum`?

Reply:
Valid point! We can use the `order by` clause to maintain the order.